### ISSUE TITLE
DataPreprocess

### ISSUE THREAD

**Issue by gkkcoding** [utomm/mini-diffuse-actor#4]
Hi，can you provide the data preprocessing steps?

**Comment by utomm** [comment]
Hi, thanks for your interest on our work.

Here we borrow the dataset generate by [3D-LUTOS](https://github.com/vlc-robot/robot-3dlotus/), with their pre-generated dataset we don't need to install RLbench for loading data and training.

As far as I know, they are following the Peract's [method](https://github.com/peract/peract?tab=readme-ov-file#data-generation) for demo collection, and add an additional downsample process. 

if you need to generate from scratch using RLbench, you can follow [their steps](https://github.com/vlc-robot/robot-3dlotus/blob/main/DATAGEN.md). However, as RLbench can not be fully controlled by seed, each run of episode or demonstration is not guarenteed to be the same, so the dataset generated might be a little different with theirs and ours.

The RLBench-18task dataset we are using (peract) can be downloaded [here](https://huggingface.co/datasets/rjgpinel/RLBench-18Task/tree/main)

**Comment by gkkcoding** [comment]
Thanks so much !I have downloaded peract data,I want to ask is the preprocess like this:generate dataset_keysteps.py -> gen_simple_policy_data.py -> get_instr_text_embeds -> get_action_text_embeds.Then I can run train/train_diffusion_policy?

**Comment by utomm** [comment]
Hi, if you need to generate from scratch using RLbench, you can follow [their steps](https://github.com/vlc-robot/robot-3dlotus/blob/main/DATAGEN.md). then you can have a folder with the exact same structure as [here](https://huggingface.co/datasets/rjgpinel/RLBench-18Task/tree/main).

If you have downloaded the peract data that we are using [here](https://huggingface.co/datasets/rjgpinel/RLBench-18Task/tree/main). you are already good to go, you don't need to do any of the policygen/keypose/embeding process. Just change the `TRAIN_DATASET.data_dir` in `minidiffuser/train/diffusion_ptv3.yaml` to where you store the dataset. and run the training :)

you can check in instruction [here](https://github.com/utomm/mini-diffuse-actor/blob/master/INSTALL.md#installation-instructions)

**Comment by gkkcoding** [comment]
Thanks so much, it works. May I ask, do you compare the result between 3D-Lotus and Mini Diffuse Actor? It seems that 3D-Lotus's memory usage is not heavy and the training time is not long. Moreover, its accuracy appears to be high from the picture you provided, but the result is not shown in the table in the paper. Could I ask the reason? Is it because the backbone is a Transformer, not a Diffusion policy?

**Comment by utomm** [comment]
Hi,

You are absolutely right—3D-Lotus achieves impressive performance without requiring significantly more computational resources. They have done excellent work, and we have a lot of respect for them.

Our code actually builds heavily upon their open-source work. To be transparent, even after several attempts at parameter tuning, we weren't able to beat their performance with our method. :(

Regarding our reporting choices:

- As you noticed we did included 3D-Lotus in the figures.

- However, for the tables, we prioritized methods that are more widely known (like RVT or Peract), those that are directly comparable to ours (like 3D-Diffuser), or the SOTA when we submitted the paper (like SAM2Act, which currently outperforms 3D-Lotus).

Here are a few of my personal insights regarding the RLBench-18 tasks and these models:

- The power of PTv3: PTv3 is a very strong encoder for 3D features. 3d-lutos is the first one to use it, as far as I know. I believe most of the performance seen in both 3D-Lotus and our method should largely thanks to that backbone.

- Classification vs. Diffusion: 3D-Lotus uses a classification-based next-pose prediction. In principle, this is often considered an "older" approach compared to diffusion heads. However, they achieve comparable (or better) results than diffusion. This suggests that for RLBench tasks, highly accurate or multi-modal continuous prediction might not be strictly necessary for good performance.

- The goal of Mini-Diffuser: In general, Mini-Diffuser behaves as expected. To ensure it stood out in at least one dimension, we optimized for training speed. If you increase the training steps, the performance does improve. However, it is difficult to "sell" a new method if it takes the same amount of time to train as 3D-Lotus just to achieve similar performance, right ;)? So we decided to offer a faster alternative which sacrifice a bit more performance.

Anyway, I'm glad to hear the training is working for you now! Please let us know if you run into any other issues.
